drivers/classic/src/display_current_settings.c

```diff
@@ -85,12 +85,12 @@ display_current_settings(int mode)
         fprintf(LOG_DEST, "EQUAL_AREA\t\tFALSE\n");
     }
     fprintf(LOG_DEST, "RESOLUTION\t\t%f\n", global_param.resolution);
-    fprintf(LOG_DEST, "TIME_STEP\t\t%d\n", global_param.dt);
-    fprintf(LOG_DEST, "SNOW_STEP\t\t%d\n", options.SNOW_STEP);
+    fprintf(LOG_DEST, "TIME_STEP\t\t%f\n", global_param.dt);
+    fprintf(LOG_DEST, "SNOW_STEP\t\t%f\n", global_param.snow_dt);
```

Comment:
Actually - a better change may be to globally replace %lf by %f if that is the correct for in C99 and C11. Either we stick with the format we have or we change them all the same way. Having a mix is what causes a maintenance headache in general. If we do replace all with %f then we do need to make sure that it works correctly with a number of different compiler (i.e. no warnings).

Bart

> On Jan 13, 2015, at 11:32 AM, Bart Nijssen <bart.nijssen@gmail.com> wrote:
> 
> Please change them. Otherwise I will. 
>> On Jan 13, 2015, at 11:29 AM, Joe Hamman <notifications@github.com> wrote:
>> 
>> In drivers/classic/src/display_current_settings.c:
>> 
>>> @@ -85,12 +85,12 @@ display_current_settings(int mode)
>>>         fprintf(LOG_DEST, "EQUAL_AREA\t\tFALSE\n");
>>>     }
>>>     fprintf(LOG_DEST, "RESOLUTION\t\t%f\n", global_param.resolution);
>>> -    fprintf(LOG_DEST, "TIME_STEP\t\t%d\n", global_param.dt);
>>> -    fprintf(LOG_DEST, "SNOW_STEP\t\t%d\n", options.SNOW_STEP);
>>> +    fprintf(LOG_DEST, "TIME_STEP\t\t%f\n", global_param.dt);
>>> +    fprintf(LOG_DEST, "SNOW_STEP\t\t%f\n", global_param.snow_dt);
>> 
>> I'm going to leave these as is. %f and %lf are both allowed by fprintf() but have identical behaviors. floats and promoted to doubles automatically. The same goes for unsigned and short ints which are promoted to int. Note, this is only true for fprintf(), the correct format specifier is important for reader functions like scanf().
>> 
>> Reference: http://stackoverflow.com/questions/4264127/correct-format-specifier-for-double-in-printf
>> 
>> —
>> Reply to this email directly or view it on GitHub.
>> 
>